### PR TITLE
Checklist validation

### PR DESCRIPTION
**Pull request by OBats**

### CODE REVIEW

File: api/controllers/checklist-controler.js [hunk 1]
```diff
@@ -1,149 +1,147 @@
...
-        const { item_title, description, details, tags, priority } = req.body;
+  try {
+    const { item_title, description, details, tags, priority } = req.body;
 
-        const list = await CheckList.findById(req.params.id);
-        const section = await list.sections_data.find(item => item.id === req.params.sectionId);
-        
-        if (!list) return res.sendStatus(404);
+    const list = await Checklist.findById(req.params.id);
+    const section = await list.sections_data.find(item => item.id === req.params.sectionId);
```
Comment (Antoniozinchenko) [inline api/controllers/checklist-controler.js]: sections_data.find is not async method
remove await

File: api/controllers/checklist-controler.js [hunk 2]
```diff
@@ -1,149 +1,147 @@
...
 
-        if (!item_title || !description || !tags || !priority) {
-            return res.status(422).json({ message: 'Please, fill up all fields' });
-        }
+    if (!list) return res.sendStatus(404);
```
Comment (Antoniozinchenko) [inline api/controllers/checklist-controler.js:36]: this line is never been invoked.. move it to 35 line before section find

**Review by romanolytvyn — APPROVED** [review]

**Review by Antoniozinchenko — CHANGES_REQUESTED** [review]

File: api/controllers/checklist-controler.js (hunk 1 above)
Comment (OBats) [reply api/controllers/checklist-controler.js]: @Antoniozinchenko Done.

File: api/controllers/checklist-controler.js (hunk 2 above)
Comment (OBats) [reply api/controllers/checklist-controler.js:36]: @Antoniozinchenko Done.

**Review by Antoniozinchenko — APPROVED** [review]

**Review by rostyslav-vasyliuk — APPROVED** [review]

**Review by Roman6917 — APPROVED** [review]

**Review by machikharostyk — APPROVED** [review]

**Review by anzhelikavelychko — APPROVED** [review]

**Pull request merged**